src/lib/y2network/interface_config_builder.rb

```diff
@@ -0,0 +1,313 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "yast"
+
+require "y2network/hwinfo"
+require "y2firewall/firewalld"
+require "y2firewall/firewalld/interface"
+
+Yast.import "LanItems"
+Yast.import "NetworkInterfaces"
+
+module Y2Network
+  # Collects data from the UI until we have enough of it to create a {Y2Network::Interface}.
+  # {Yast::LanItemsClass#Commit Yast::LanItems.Commit(builder)} uses it.
+  class InterfaceConfigBuilder
```

Comment:
Maybe nitpicking, but shouldn't it be called `InterfaceBuilder` then? Because a `ConnectionConfigBuilder` builds a `ConnectionConfig`, not a `Connection`.